sled-agent/src/bootstrap/agent.rs

```diff
@@ -89,74 +125,84 @@ impl Agent {
-                if other_agents.len() < UNLOCK_THRESHOLD - 1 {
-                    warn!(&self.log, "Not enough peers to start establishing quorum");
+                if other_agents.len() < share().threshold - 1 {
+                    warn!(
+                        &self.log,
+                        "Not enough peers to start establishing quorum"
+                    );
                     return Err(BackoffError::Transient(
                         BootstrapError::NotEnoughPeers,
                     ));
                 }
-                info!(&self.log, "Bootstrap: Enough peers to start share transfer");
-
-                // TODO-correctness:
-                // - Establish trust quorum.
-                // - Once this is done, "unlock" local storage
-                //
-                // The current implementation sends a stub request to all known sled
-                // agents, but does not actually create a quorum / unlock anything.
-                let other_agents: Vec<BootstrapClient> = other_agents
+                info!(
+                    &self.log,
+                    "Bootstrap: Enough peers to start share transfer"
+                );
+
+                // Retrieve verified rack_secret shares from a quorum of agents
+                let other_agents: Vec<trust_quorum::Client> = other_agents
                     .into_iter()
                     .map(|mut addr| {
-                        addr.set_port(BOOTSTRAP_PORT);
-                        // TODO-correctness:
-                        //
-                        // Many rust crates - such as "URL" - really dislike
-                        // using scopes in IPv6 addresses. Using
-                        // "addr.to_string()" results in an IP address format
-                        // that is rejected when embedded into a URL.
-                        //
-                        // Instead, we merely use IP and port for the moment,
-                        // which loses the scope information. Longer-term, if we
-                        // use ULAs (Unique Local Addresses) the scope shouldn't
-                        // be a factor anyway.
-                        let addr_str = format!("[{}]:{}", addr.ip(), addr.port());
-                        info!(&self.log, "bootstrap: Connecting to {}", addr_str);
-                        BootstrapClient::new(
-                            &format!("http://{}", addr_str),
-                            self.log.new(o!(
-                                "Address" => addr_str,
-                            )),
+                        addr.set_port(trust_quorum::PORT);
+                        trust_quorum::Client::new(
+                            &self.log,
+                            share().verifier.clone(),
+                            addr,
                         )
                     })
                     .collect();
+
+                // TODO: Parallelize this and keep track of who's shares we've already retrieved and
```

Comment:
mega-nitpick

```suggestion
                // TODO: Parallelize this and keep track of whose shares we've already retrieved and
```

Reply:
Oof, thanks for finding this typo! Fixed.